Geometry/HGCalCommonData/plugins/dd4hep/DDHGCalEEAlgo.cc

```diff
@@ -0,0 +1,416 @@
+            double rmax = routF * cosAlpha_ - tol;
+            pgonZ.emplace_back(-hthick);
+            pgonZ.emplace_back(hthick);
+            pgonRin.emplace_back(rinB);
+            pgonRin.emplace_back(rinB);
+            pgonRout.emplace_back(rmax);
+            pgonRout.emplace_back(rmax);
+          } else {
+            HGCalGeomTools::radius(zz - hthick,
+                                   zz + hthick,
+                                   zFrontB_,
+                                   rMinFront_,
+                                   slopeB_,
+                                   zFrontT_,
+                                   rMaxFront_,
+                                   slopeT_,
+                                   -layerSense_[ly],
+                                   pgonZ,
+                                   pgonRin,
+                                   pgonRout);
+            for (unsigned int isec = 0; isec < pgonZ.size(); ++isec) {
+              pgonZ[isec] -= zz;
+              pgonRout[isec] = pgonRout[isec] * cosAlpha_ - tol;
+            }
+          }
+
+          dd4hep::Solid solid =
+              dd4hep::Polyhedra(sectors_, -alpha_, 2. * cms_units::piRadians, pgonZ, pgonRin, pgonRout);
+          ns.addSolidNS(ns.prepend(name), solid);
+          glog = dd4hep::Volume(solid.name(), solid, matter);
+          ns.addVolumeNS(glog);
+
+#ifdef EDM_ML_DEBUG
+          edm::LogVerbatim("HGCalGeom") << "DDHGCalEEAlgo: " << solid.name() << " polyhedra of " << sectors_
+                                        << " sectors covering " << convertRadToDeg(-alpha_) << ":"
+                                        << convertRadToDeg(-alpha_ + 2._pi) << " with " << pgonZ.size()
+                                        << " sections and filled with " << matName;
+
+          for (unsigned int k = 0; k < pgonZ.size(); ++k)
+            edm::LogVerbatim("HGCalGeom")
+                << "[" << k << "] z " << pgonZ[k] << " R " << pgonRin[k] << ":" << pgonRout[k];
+#endif
+        } else {
+          dd4hep::Solid solid = dd4hep::Tube(rinB, routF, hthick, 0.0, 2. * cms_units::piRadians);
+          ns.addSolidNS(ns.prepend(name), solid);
+          glog = dd4hep::Volume(solid.name(), solid, matter);
+          ns.addVolumeNS(glog);
+
+#ifdef EDM_ML_DEBUG
+          edm::LogVerbatim("HGCalGeom") << "DDHGCalEEAlgo: " << solid.name() << " Tubs made of " << matName
+                                        << " of dimensions " << rinB << ", " << routF << ", " << hthick
+                                        << ", 0.0, 360.0 and position " << glog.name() << " number " << copy << ":"
+                                        << layerCenter_[copy - 1];
+#endif
+          PositionSensitive(ctxt, e, glog, rinB, routF, zz, layerSense_[ly], layerCenter_[copy - 1]);  //, cpv);
+        }
+
+        dd4hep::Position r1(0, 0, zz);
+        dd4hep::Rotation3D rot;
+        mother_.placeVolume(glog, copy, dd4hep::Transform3D(rot, r1));
```

Comment:
According to #27997, it looks like the `Rotation3D` is not needed. Have you seen that PR?

Reply:
Yes - now I removed this